base/server/healthcheck/pki/server/healthcheck/clones/connectivity_and_data.py

```diff
@@ -0,0 +1,210 @@
+# Authors:
+#  Jack Magne    <jmagne@redhat.com> #
+# Copyright Red Hat, Inc.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+import logging
+
+from pki.server.healthcheck.clones.plugin import ClonesPlugin, registry
+from pki.client import PKIConnection
+from pki.cert import CertClient
+from ipahealthcheck.core.plugin import Result, duration
+from ipahealthcheck.core import constants
+
+logger = logging.getLogger(__name__)
+
+
+@registry
+class ClonesConnectivyAndDataCheck(ClonesPlugin):
+    """
+    Assure master and clones within a  pki instance are reachable
+    """
+    def check_ca_clones(self):
+        for host in self.clone_cas:
+            cur_clone_msg = ' Host: ' + host.Hostname + ' Port: ' + host.SecurePort
+            # Reach out and get some certs, to serve as a data and connectivity check
+            try:
+                connection = PKIConnection(protocol='https',
+                                           hostname=host.Hostname,
+                                           port=host.SecurePort,
+                                           verify=False)
+
+                cert_client = CertClient(connection)
+                # get the first 3 in case we cant to make a sanity check of replicated data
+                certs = cert_client.list_certs(size=3)
+
+                if certs is not None and len(certs.cert_data_info_list) == 3:
+                    logger.info('Cert data successfully obtained from clone.')
+                else:
+                    raise BaseException('CA clone problem reading data.' + cur_clone_msg)
+            except BaseException as e:
+                logger.error("Internal server error %s", e)
+                raise BaseException('Internal error testing CA clone.' + cur_clone_msg)
+
+        return
```

Comment:
(nitpick) I think this line is unnecessary?

Reply:
I agree with the modularity point, but the checks, the way they are arranged now, do return separate results for each type of subsystem. But yes, the refactor seems quite reasonable.